lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -87,9 +92,9 @@ namespace Js
         handler = VarTo<DynamicObject>(args[2]);
         if (VarIs<JavascriptProxy>(handler))
         {
-            if (VarTo<JavascriptProxy>(handler)->handler == nullptr)
+            if (VarTo<JavascriptProxy>(handler)->IsRevoked())
```

Comment:
Change this VarTo to an UnsafeVarTo. (VarTo includes a check but we've already done the check in the if on line 93)

(Unrelated to the change you're doing but a good drive by improvement to add)

Reply:
Done, and I've rebased it from the latest master branch.